frontend-new/src/sensitiveData/components/sensitiveData/SensitiveData.tsx

```diff
@@ -0,0 +1,276 @@
+
+  const handleSaveSensitivePersonalData = useCallback(async () => {
+    setIsSavingSensitiveData(true);
+
+    try {
+      const userPreferences = userPreferencesStateService.getUserPreferences();
+
+      if (!userPreferences) {
+        // something is not right, we should have user preferences at this point.
+        throw new Error("User preferences not found.");
+      }
+
+      console.log({ userPreferences });
+      await sensitivePersonalDataService.createSensitivePersonalData(sensitivePersonalData, userPreferences.user_id);
+
+      // Update user preferences to indicate that the user has sensitive personal data
+      // so that the user is not prompted to provide this information again.
+      // We set the state directly because we don't want to go to the server to get the updated the user preferences.
+      userPreferencesStateService.setUserPreferences({
+        ...userPreferences,
+        has_sensitive_personal_data: true,
+      });
+
+      enqueueSnackbar("Personal data saved successfully and securely.", { variant: "success" });
+      navigate(routerPaths.ROOT);
+    } catch (e) {
+      if (e instanceof ServiceError) {
+        writeServiceErrorToLog(e, console.error);
+        enqueueSnackbar(getUserFriendlyErrorMessage(e), { variant: "error" });
+      } else {
+        enqueueSnackbar(`Failed to save personal data: ${(e as Error).message}`, { variant: "error" });
+        console.error("Failed to save personal data", e);
+      }
+    } finally {
+      setIsSavingSensitiveData(false);
+    }
+  }, [enqueueSnackbar, navigate, sensitivePersonalData]);
+
+  const handleRejectProvidingSensitiveData = useCallback(async () => {
+    try {
+      const authenticationService = AuthenticationServiceFactory.getCurrentAuthenticationService();
+      await authenticationService!.logout();
+      navigate(routerPaths.LOGIN, { replace: true });
+      enqueueSnackbar("Successfully logged out.", { variant: "success" });
+    } catch (e) {
+      console.error("Failed to log out", e);
+      enqueueSnackbar("Failed to log out.", { variant: "error" });
+    }
+  }, [enqueueSnackbar, navigate]);
+
+  return (
+    <>
+      <Container maxWidth="xs" sx={{ height: "100%" }} data-testid={DATA_TEST_ID.SENSITIVE_DATA_CONTAINER}>
+        <Box
+          display="flex"
+          flexDirection="column"
+          alignItems="center"
+          gap={theme.fixedSpacing(theme.tabiyaSpacing.lg)}
+          width={"100%"}
+        >
+          <AuthHeader
+            title={"Provide Your Information"}
+            subtitle={
+              "Provide your personal details carefully. Your information is secure and cannot be changed after submission."
+            }
+          />
+
+          <Box width={"100%"} display={"flex"} flexDirection={"column"} gap={theme.tabiyaSpacing.xl}>
+            <Box display="flex" flexDirection="column" gap={theme.tabiyaSpacing.lg}>
+              <TextField
```

Comment:
since we have an object and the only difference between the multiple fields is the label, what about including the label as a field in the fieldFormProps object and creating the TextFields using a loop?